orchestrator/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,480 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+
+	"github.com/rancher/longhorn-manager/crdstore"
+	"github.com/rancher/longhorn-manager/orchestrator"
+	"github.com/rancher/longhorn-manager/types"
+	"github.com/rancher/longhorn-manager/util"
+
+	apiv1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kCli "k8s.io/client-go/kubernetes"
+)
+
+const (
+	cfgDirectory = "/var/lib/rancher/longhorn/"
+	hostUUIDFile = cfgDirectory + ".physical_host_uuid"
+)
+
+var (
+	WaitDeviceTimeout = 30 //seconds
+	WaitAPITimeout    = 30 //seconds
+	WaitPodTimeout    = 5  //seconds
+	WaitPodPeriod     = 3  //seconds
+	WaitPodCounter    = 5
+)
+
+type Kubernetes struct {
+	EngineImage string
+	IP          string
+	currentNode *types.NodeInfo
+	cli         *kCli.Clientset
+}
+
+type Config struct {
+	EngineImage string
+}
+
+func NewKuberOrchestrator(cfg *Config) (*Kubernetes, error) {
+	if cfg.EngineImage == "" {
+		return nil, fmt.Errorf("missing required parameter EngineImage")
+	}
+
```

Comment:
Check for all needed environment variables before getting into the function calls. It will be too late them.

Reply:
And use constants for the environement variables.